### DISCUSSION TITLE
restrict dock applications for each desktop or viewport

### DISCUSSION THREAD

**Discussion by skypce** [linuxdeepin/developer-center#3061] [Features Request & Ideas | 特性请求 & 头脑风暴]
Hi, will be great if the dock can separate applications by viewport or desktop.
Example if i am in Desktop 2 i dont want see in dock applications from Desktop 1.
If i am in desktop 1 , i dont want see applications from desktop 2 or 3.

Example 2:

If i click in google chrome in desktop 1 , i dont want see google chrome from desktop 2.
if i click in google chrome in desktop 2, i dont want see the google chrome from desktop 1.

When i click in dock app " show all windows " it must look the windows of the app of current desktop not all.
You can have a " show all windows of current desktop" and "show all windows from desktop 1" "show all windows from desktop 2 " "show all windows from desktop 3"
I really apreciate if you fix it , will be really great.

**Comment by Grafikart** [comment]
I would expand this enhancement with the support of multiple screens. It would be awesome to be able to expand the bar (or clone the dock) on every screen, showing only the windows visible on this screen.

It would be nice to have these options :
- Show all applications
- Only show applications visible in the current workspace
- Only show applications visible in the current screen and workspace.

@skypce "show all windows of current desktop" is available already, you can use a shortcut to do it. But not through the dock indeed.

**Comment by Fuhrmann** [comment]
That would be nice to have!

**Comment by ssk-wh** [comment]
Good idea, but what about the tray icon? Any good suggestions?
